### DISCUSSION TITLE
Check for 32 or 64 build

### DISCUSSION THREAD

**Discussion by sanumesh** [openssl/openssl#24507] [Q&A] [answered]
Hello

How to check in the source code if the compilation is happening for a 32-bit or 64-bit ?
Basically I want to add a fix for 32-bit build in openssl poll code (ssl/rio/poll_immediate.c). 
The fix needs to be enabled only when configure with aix-cc and not for aix64-cc case. Which flag can I use to add this fix ?

Thanks

Regards
Sandeep Umesh
AIX Security Development

**Comment by nhorman** [comment, marked as answer]
can't you just use the compiler macros for that?

i.e. on x86_64 with gcc building for 64 bit implicitly defines __x86_64__

ostensibly the aix compiler defines something similar for ppc?
